tests/base_tests.py

```diff
@@ -585,7 +585,7 @@ def test_fit_not_changing_data_weighted(
     @pytest.mark.parametrize(
         "bad_weight_value, expected_message",
         [
-            (np.nan, "weight column must be non-null"),
```

Comment:
Same issue as in #349 that removing np.nan for all cases is perhaps not the best idea? @Chip2916 is looking into adding handling in median imputer for np.nan which would fix this, might be able to do simillar here and have both None and np.nan cases in this base test. https://github.com/lvgig/tubular/pull/349#discussion_r1861964504

If we cant do np.nan then we should probably juts be overriding for this transformers case rather than changing for the whole package.

Reply:
ahh yes good point, want to test both 

I've added handling here that I think will work until the narwhals method becomes available?

https://github.com/lvgig/tubular/pull/345/commits

Reply:
https://github.com/lvgig/tubular/pull/345/commits/30061ed3bbf6d05100eaddb8a3ade23bc5176b48 looks like it addresses this adequately until is_nan gets implemented in narwhals.